Bottom: 754be25be7f1808e265bf2e8a65d190fdeaa75da
Top:    6fb3405f1d68a4ed924959315aa8a24e0da1f8b9
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-06-05 12:12:39 +0800

Refresh of vhost-allow-to-disable-device

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 042030e..03c0572 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1566,12 +1566,14 @@ EXPORT_SYMBOL_GPL(vhost_vring_ioctl);
 
 int vhost_init_device_iotlb(struct vhost_dev *d, bool enabled)
 {
-	struct vhost_umem *niotlb, *oiotlb;
+	struct vhost_umem *niotlb = NULL, *oiotlb;
 	int i;
 
-	niotlb = vhost_umem_alloc();
-	if (!niotlb)
-		return -ENOMEM;
+	if (enabled) {
+		niotlb = vhost_umem_alloc();
+		if (!niotlb)
+			return -ENOMEM;
+	}
 
 	oiotlb = d->iotlb;
 	d->iotlb = niotlb;
